### PR TITLE
Add support for grad_fn on pointers

### PR DESCRIPTION
**Pull request by LaRiffle**
## Description
We need that wrapper.grad_fn refer to the ptr.grad_fn they wrap!

Implem is very close to .grad

**Is needed for the Opacus / PySyft demo**

### CODE REVIEW

File: syft/frameworks/torch/tensors/interpreters/native.py [hunk 1]
```diff
@@ -189,6 +189,27 @@ def data(self):
...
+        if new_grad_fn is not None and (
+            not isinstance(new_grad_fn, torch.Tensor) or hasattr(new_grad_fn, "child")
+        ):
+            self.child.grad_fn = new_grad_fn
+        else:
+            if hasattr(self, "native_grad_fn"):
+                with torch.no_grad():
+                    self.native_grad_fn = new_grad_fn
+            elif new_grad_fn is not None:
+                self.native_grad_fn = new_grad_fn
```
Comment (gmuraru) [inline syft/frameworks/torch/tensors/interpreters/native.py:210]: Q: Is there any way we could add tests for the codecov issues?

Comment (LaRiffle) [reply syft/frameworks/torch/tensors/interpreters/native.py:210]: @gmuraru done :)

**Review by gmuraru — APPROVED** [review]
LGTM!

**Review by youben11 — APPROVED** [review]
LGTM!

**Pull request merged**